t/v3/health_check.t

```diff
@@ -0,0 +1,438 @@
+
+            etcd:set("/fault_count", { a='abc'})
+            etcd:set("/fault_count", { a='abc'})
+            local fails, err = ngx.shared["etcd_cluster_health_check"]:get("http://127.0.0.1:42379")
+            if err then
+                ngx.say(err)
+            end
+            ngx.say(fails)
+        }
+    }
+--- request
+GET /t
+--- response_body
+2
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: check endpoint is healthy
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua_block {
+            local health_check, err = require "resty.etcd.health_check" .init({
+                shm_name = "etcd_cluster_health_check",
+                fail_timeout = 3,
+                max_fails = 1,
+            })
+
+            local etcd, err = require "resty.etcd" .new({
+                protocol = "v3",
+                http_host = {
+                    "http://127.0.0.1:42379",
+                    "http://127.0.0.1:22379",
+                    "http://127.0.0.1:32379",
+                },
+                user = 'root',
+                password = 'abc123',
+            })
+
+            etcd:set("/get_target_status", { a='abc'})
+
+            local healthy = health_check.get_target_status("http://127.0.0.1:42379")
+            ngx.say(healthy)
+        }
+    }
+--- request
+GET /t
+--- response_body
+false
+--- no_error_log
+[error]
+
+
+
+=== TEST 7: make sure `fail_timeout` works
+--- http_config eval: $::HttpConfig
+--- config
+    location /t {
+        content_by_lua_block {
+            local health_check, err = require "resty.etcd.health_check" .init({
+                shm_name = "etcd_cluster_health_check",
+                fail_timeout = 2,
+                max_fails = 1,
+            })
+
+            local etcd, err = require "resty.etcd" .new({
+                protocol = "v3",
+                http_host = {
+                    "http://127.0.0.1:42379",
+                    "http://127.0.0.1:22379",
+                    "http://127.0.0.1:32379",
+                },
+                user = 'root',
+                password = 'abc123',
+            })
+
+            local res, err = etcd:set("/fail_timeout", "http://127.0.0.1:42379")  -- trigger http://127.0.0.1:42379 to unhealthy
+
+            res, err = etcd:set("/fail_timeout", "http://127.0.0.1:22379") -- choose http://127.0.0.1:22379 to set value
+            res, err = etcd:get("/fail_timeout")
+            assert(res.body.kvs[1].value == "http://127.0.0.1:22379")
+
+            ngx.sleep(2)
+
+            res, err = etcd:set("/fail_timeout", "http://127.0.0.1:42379") -- choose http://127.0.0.1:42379 to set value
+            res, err = etcd:get("/fail_timeout")
+            assert(res.body.kvs[1].value == "http://127.0.0.1:22379")
+
+            ngx.say("done")
+        }
+    }
+--- request
+GET /t
+--- timeout: 5
+--- response_body
+done
+--- no_error_log
+[error]
+
+
+
+=== TEST 8: has no healthy etcd endpoint, follow old style
```

Comment:
I didn't get why when no endpoints are healthy, we should follow old style. Since as #101 talked about
> When all instances of a certain api fail (such as auth api), it will cause crazy retries, which may eventually overwhelm the ETCD cluster.
Following old styles still could not solve this problem

Reply:
I can't refute this example.
we do not request etcd when all etcd instances are down.
what do you think @membphis ?

Reply:
IMO maybe we could leave some warnings and like add some delay for continuous retries? I'm not so sure.
Hi @nic-chen do you have some opinions on this

Reply:
ping @membphis @nic-chen

Reply:
ping @membphis @nic-chen

Reply:
sorry I missed this message. We should fix it in a new PR. 

@Yiyiyimu had created the issue https://github.com/api7/lua-resty-etcd/issues/111 about this comment.

many thx